frontend/src/App.tsx

```diff
@@ -1,8 +1,37 @@
-import Router from './routes';
+// import Login from './pages/Login';
+
+import React from 'react';
 import { RouterProvider } from 'react-router-dom';
+import { IAuth, IAuthContext } from './interfaces';
+import router from './routes';
+import { useLocalStorageState } from './utils/hooks';
 
-function App() {
-  return <RouterProvider router={Router} />;
+export default function App() {
+  return (
+    <AuthProvider>
+      <RouterProvider router={router} />
+    </AuthProvider>
+  );
 }
 
-export default App;
+export const AuthContext = React.createContext<IAuthContext>(null!);
+
+function AuthProvider({ children }: { children: React.ReactNode }) {
+  const [authState, setAuthState] = useLocalStorageState<IAuth | null>('authState', null);
+
+  const redirectToSignIn = () => {
+    window.location.replace(`${import.meta.env.VITE_BACKEND_API_HOST}/v1/auth/google`);
+  };
+
+  const signout = () => {
+    setAuthState(null);
+  };
+
+  const signIn = (auth: IAuth) => {
+    setAuthState(auth);
+  };
+
+  const value = { authState, redirectToSignIn, signIn, signout };
+
+  return <AuthContext.Provider value={value}>{children}</AuthContext.Provider>;
+}
```

Comment:
Let's abstract these away into another file(s), together with the interfaces. Maybe under frontend/providers or frontend/contextProviders